warehouse/integrations/deltalake-native/deltalake_native.go

```diff
@@ -0,0 +1,1367 @@
+		if partitionQuery, err = d.partitionQuery(warehouseutils.UsersTable); err != nil {
+			return map[string]error{
+				warehouseutils.IdentifiesTable: nil,
+				warehouseutils.UsersTable:      fmt.Errorf("getting partition query: %w", err),
+			}
+		}
+
+		pk := primaryKey(warehouseutils.UsersTable)
+
+		query = fmt.Sprintf(`
+			MERGE INTO %[1]s.%[2]s AS MAIN USING (
+			  SELECT
+				%[6]s
+			  FROM
+				%[1]s.%[3]s
+			) AS STAGING ON %[8]s MAIN.%[4]s = STAGING.%[4]s
+			WHEN MATCHED THEN
+			UPDATE
+			SET
+			  %[5]s WHEN NOT MATCHED
+			THEN INSERT (%[6]s)
+			VALUES
+			  (%[7]s);
+		`,
+			d.Namespace,
+			warehouseutils.UsersTable,
+			stagingTableName,
+			pk,
+			columnsWithValues(columnKeys),
+			columnNames(columnKeys),
+			stagingColumnNames(columnKeys),
+			partitionQuery,
+		)
+	}
+
+	row = d.DB.QueryRow(query)
+
+	if row.Err() != nil {
+		return map[string]error{
+			warehouseutils.IdentifiesTable: nil,
+			warehouseutils.UsersTable:      fmt.Errorf("running deduplication: %w", row.Err()),
+		}
+	}
+
+	var (
+		affected int64
+		updated  int64
+		deleted  int64
+		inserted int64
+	)
+
+	if d.LoadTableStrategy == appendMode {
+		err = row.Scan(&affected, &inserted)
+	} else {
+		err = row.Scan(&affected, &updated, &deleted, &inserted)
+	}
+
+	if err != nil {
+		return map[string]error{
+			warehouseutils.IdentifiesTable: nil,
+			warehouseutils.UsersTable:      fmt.Errorf("getting rows affected for dedup: %w", err),
+		}
+	}
+
+	d.Stats.NewTaggedStat("dedup_rows", stats.CountType, stats.Tags{
+		"sourceID":    d.Warehouse.Source.ID,
+		"sourceType":  d.Warehouse.Source.SourceDefinition.Name,
+		"destID":      d.Warehouse.Destination.ID,
+		"destType":    d.Warehouse.Destination.DestinationDefinition.Name,
+		"workspaceId": d.Warehouse.WorkspaceID,
+		"namespace":   d.Namespace,
+		"tableName":   warehouseutils.UsersTable,
+	}).Count(int(updated))
+
+	d.Logger.Infow("completed loading for users and identifies tables",
+		logfield.SourceID, d.Warehouse.Source.ID,
+		logfield.SourceType, d.Warehouse.Source.SourceDefinition.Name,
+		logfield.DestinationID, d.Warehouse.Destination.ID,
+		logfield.DestinationType, d.Warehouse.Destination.DestinationDefinition.Name,
+		logfield.WorkspaceID, d.Warehouse.WorkspaceID,
+		logfield.Namespace, d.Namespace,
+	)
+
+	return map[string]error{
+		warehouseutils.IdentifiesTable: nil,
+		warehouseutils.UsersTable:      nil,
+	}
+}
+
+func (d *Deltalake) tableLocationQuery(tableName string) string {
+	var (
+		enableExternalLocation = warehouseutils.GetConfigValueBoolString(EnableExternalLocation, d.Warehouse)
+		externalLocation       = warehouseutils.GetConfigValue(ExternalLocation, d.Warehouse)
+	)
+
+	if enableExternalLocation != "true" {
+		return ""
+	}
+	if externalLocation == "" {
+		return ""
+	}
+	return fmt.Sprintf("LOCATION '%s/%s/%s'", externalLocation, d.Namespace, tableName)
+}
+
+func (d *Deltalake) dropDanglingStagingTables() {
+	tableNames, _ := d.fetchTables("^rudder_staging_.*$")
```

Comment:
isn't dangerous to ignore error here. Since, there is a possibility that there is nothing in `tableNames` because some error occured while fetching the tables names. This will led to the accumulation of those tables. I believe we should atleast log it.

Reply:
Good point. let me add error logs here.

Reply:
Although, we are dropping staging tables after every sync. But there might be a case where we didn't have DROP permissions and since we are skipping these errors, these tables will accumulate.

Reply:
Added warning logs for this.